File: exercism/go/pangram/pangram.go
package pangram

import "unicode"

func IsPangram(s string) bool {
	var letterSet = make(map[rune]bool)
	for _, r := range "abcdefghijklmnopqrstuvwxyz" {
		letterSet[r] = true
	}
	for _, r := range s {
		delete(letterSet, unicode.ToLower(r))
	}
	return len(letterSet) == 0
}

File: exercism/c/pangram/src/pangram.c
#include <stdint.h>
#include "pangram.h"
#include <ctype.h>

bool is_pangram(const char *sentence) {
    if (sentence == NULL) {
        return false;
    }
    bool chars[26] = {false};
    char c;

    while ((c = *sentence++) != '\0') {
        if (isalpha(c)) {
            int idx = tolower(c) - 'a';
            chars[idx] = true;
        }
    }

    for (int i = 0; i < 26; i++) {
        if (!chars[i]) {
            return false;
        }
    }

    return true;
}
